f8a_notification/entryscript.py

```diff
@@ -14,14 +14,286 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-# Author: Geetika Batra <gbatra@redhat.com>
+# Author: Yusuf Zainee <yzainee@redhat.com>
 #
 
 """Class to create cron script."""
 
 
+import json
+from pprint import pprint
+from datetime import datetime, timedelta
+from utils import select_latest_version, execute_gremlin_dsl
+from utils import get_response_data, check_license_conflict
+from uuid import uuid4
+import logging
+
+PACKAGE_DATA = {}
+REPO_DATA = {}
+NEW_VERSION_DATA = {}
+VERSION_DATA = {}
+FINAL_DATA = {}
+
+logger = logging.getLogger(__name__)
+
+
+def get_value(json, property):
+    """Get the values from json."""
+    return json.get(property, [''])[0]
+
+
 def run():
-    print("Hello World")
+    """Entrypoint function."""
+    logger.info("Scheduled scan for newer versions started")
+    read_packages()
+    remove_cve_versions()
+    get_repos()
+    find_latest_version()
+    check_license_compatibility()
+    generate_notification_payload()
+    logger.info("Scheduled scan for newer versions finished")
+
+
+def read_packages():
+    """Read all the packages last updated."""
+    prev_date = (datetime.utcnow() - timedelta(1)).strftime('%Y%m%d')
+    query_str = "g.V().has('latest_version_last_updated',prev_date).valueMap()"
+    # prev_date = '20180805' for testing purpose, change date here
+    payload = {
+        'gremlin': query_str,
+        'bindings': {
```

Comment:
Nice! I didn't know this is possible. We do it "the weird way" in other services.

Reply:
Since very long - https://github.com/fabric8-analytics/f8a-server-backbone/blob/master/src/recommender.py#L139